CMakeLists.txt

```diff
@@ -120,8 +127,11 @@ set_target_properties(${APP_NAME} PROPERTIES
                       XCODE_ATTRIBUTE_TARGETED_DEVICE_FAMILY ${DEVICE_FAMILY}
                       MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/plist.in
                       XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES
-                      XCODE_ATTRIBUTE_COMBINE_HIDPI_IMAGES "NO"
-                      XCODE_ATTRIBUTE_INSTALL_PATH "$(LOCAL_APPS_DIR)")
+                      XCODE_ATTRIBUTE_COMBINE_HIDPI_IMAGES NO
+                      XCODE_ATTRIBUTE_INSTALL_PATH "$(LOCAL_APPS_DIR)"
+                      XCODE_ATTRIBUTE_ENABLE_TESTABILITY YES
+                      XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN YES
```

Comment:
I do not see link-time error you say you have? I have Xcode 8.3.3 and Simulator 9.0 and 10.3.1...

As for XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN I'm not 100% sure but I seem to recall that this applies visibility to all supported languages not just C++ e.g. ObjC. Anyhow this is default for app targets in Xcode so it makes sense to have this specified there.